README.md

```diff
@@ -1 +1,42 @@
+
+## 1. Proposal and technical document
+
+To ensure a clear connection between proposals and their corresponding tasks, a dedicated folder for each proposal will be created within this repository. The folder name will be the unique JIRA issue associated with the RFE. This JIRA issue should be opened before creating the proposal.
+
+In order to facilitate open discussions and transparent collaboration, a pull request for the technical document must be raised for each proposal. The discussion will thus evolve through reviews and feedback. When the proposal is given the go-ahead by the reviewers, the pull request will be merged.
+
+Once the technical document has been approved and merged, it cannot be modified. If any improvement or modification is needed, the new technical document will have to be reviewed and approved by all reviewers.
+
+To facilitate the navigation between the pull request and the related JIRA, a hyperlink to the JIRA should be added to the description of the pull request and, vice versa, a link to the pull request should be attached to the JIRA.
+
+## 2. Implementation Document (optional)
+
+An optional implementation-specific document can be submitted at this stage. This document should contain a clear roadmap to ensure a smooth transition from design to implementation. Although this step is optional, it might become necessary when there are disagreements among reviewers. Any time there is a disagreement on the implementation of the technical document, a pull request for an implementation-specific document must be raised and a discussion should be initiated. Once the Narayana team agrees on a specific implementation, the pull request should be merged and the RFE must be developed based on that document.
+
+Once the implementation-specific document has been approved and merged, it cannot be modified. If any improvement or modification is needed, the new implementation-specific document will have to be reviewed and approved by all reviewers.
+
+To facilitate the navigation between the pull request and the related JIRA, a hyperlink to the JIRA should be added to the description of the pull request and, vice versa, a link to the pull request should be attached to the JIRA.
+
+## 3. Pull Request
+
+Once the technical document and the optional implementation-specific document are approved and merged, community members are free to propose a pull request that modifies the code base of Narayana in accordance with the technical and the implementation-specific documents. Pull requests should be merged only when they are fully compliant with the aforementioned documents and are passing all tests of the Narayana CI. The usual review process is still needed before merging the pull request but reviews should not address the overall design or implementation of the RFE.
```

Comment:
Not being able to comment on the implementation of the RFE is going to be a problem since you cannot capture the implementation in a document because implementation is code.

Reply:
I disagree. The technical document is a common approach in the software industry to discuss RFEs. The software engineer presenting the RFE and reviewers should be able to work together to discuss the overall design of the RFE before the approval of the technical document. Moreover, the workflow presented here gives our community the possibility to submit an implementation document to discuss technical details of the implementation. I think that the technical document together with the implementation document should give the community enough opportunities to discuss at a satisfactory level all the design details of an RFE.

Reply:
I think seeing a proposed format for implementation documents would be helpful with this discussion

Reply:
I was referring to the implementation document only. If we cannot comment on the implementation then what will be the purpose of the PR that ultimately adds the feature. The only way that you could produce a valid implementation document that cannot be discussed during the implementation phase would be to use a formal specification language.

Reply:
Possibly: `The usual review process is still needed before merging the pull request but reviews should not address the overall design or implementation of the RFE` -> `The usual review process is still needed before merging the pull request but reviews should not address the overall design or implementation choices that are approved in the proposals document`

Reply:
I have suggested some changes at https://github.com/jmfinelli/narayana-proposals/pull/1 including a change to clarify something of the implementation of the RFE review (slight change to what I mentioned for accuracy).

Reply:
Thanks for your PR, Tom. I have resolved this discussion. Feel free to unresolve it if you need to continue the discussion.